src/v/redpanda/tests/fixture.h

```diff
@@ -789,7 +789,7 @@ class redpanda_thread_fixture {
         { r.encode(writer, version) } -> std::same_as<void>;
     }
     kafka::request_context make_request_context(
-      RequestType request, kafka::request_header& header, conn_ptr conn = {}) {
+      RequestType& request, kafka::request_header& header, conn_ptr conn = {}) {
```

Comment:
Can you give some detail here? This doesn't strike me as a safe change without some additional information, since callers may now see the request they are passing in mutated, as opposed to having the copy mutated (leaving the caller non-the-wiser).

Reply:
Yeah, you're right that this is a risk. I checked the code paths and didn't see any modifications to the request to thought this would be safe (though admittedly this doesn't guard against future changes).

Changing to a reference was necessary because `partition_produce_data` has a deleted default copy constructor due to its `records` field, e.g. `std::optional<kafka::produce_request_record_data> records{}' and unlike `kafka::fetch_request` it's not possible to copy a `kafka::produce_request` as-is. Would you prefer to see a user-defined copy constructor for `partition_produce_data` ?

Reply:
If it is not modified, we could just make it `const`? 

However, I think the second part of your reply is suggesting it is modified: do we move out of `request` in order to avoid copying? "Moving out of" is a modification.

So I think if it is truly not modified, make it `const`. If it is modified, we should try to leave the signature alone move it at the call site.

Reply:
Of course you're right, it is modified :) 

Switched to doing a move at the call site.